sdks/java/core/src/main/java/org/apache/beam/sdk/values/RowWithGetters.java

```diff
@@ -122,6 +123,15 @@ private Iterable getIterableValue(FieldType elementType, Object fieldValue) {
                   cacheKey, i -> getMapValue(type.getMapKeyType(), type.getMapValueType(), map))
           : (T) getMapValue(type.getMapKeyType(), type.getMapValueType(), map);
     } else {
+      if (type.isLogicalType(OneOfType.IDENTIFIER)) {
```

Comment:
I thought that originally, however it turns out that the only way to make OneOf work with the getters framework is to actually handle it explicitly, otherwise recursive translation fails to work. Happy to pull this out into a separate PR if you think that's better.

Reply:
My gut feeling says that we should take it separately. Ok, union types will fail, but for now it's only proto specific right? I've added this on my todo list when testing the implementation with our use cases.